ethereum-consensus/src/capella/block_processing.rs

```diff
@@ -252,16 +255,113 @@ pub fn process_withdrawals<
         MAX_WITHDRAWALS_PER_PAYLOAD,
     >,
-    _context: &Context,
+    context: &Context,
 ) -> Result<()> {
-    unimplemented!()
+    let expected_withdrawals = get_expected_withdrawals(state, context);
+
+    if execution_payload.withdrawals.as_ref() == expected_withdrawals {
+        return Err(invalid_operation_error(InvalidOperation::Withdrawal(
+            InvalidWithdrawal::IncorrectWithdrawals {
+                expected: expected_withdrawals.len(),
+                count: execution_payload.withdrawals.len(),
+            },
+        )))
+    }
+
+    for withdrawal in &expected_withdrawals {
+        decrease_balance(state, withdrawal.validator_index, withdrawal.amount);
+    }
+
+    // Update the next withdrawal index if this block contained withdrawals
+    if let Some(latest_withdrawal) = expected_withdrawals.last() {
+        state.next_withdrawal_index = latest_withdrawal.index + 1;
+    }
+
+    // Update the next validator index to start the next withdrawal sweep
+    if expected_withdrawals.len() == context.max_withdrawals_per_payload {
+        // Next sweep starts after the latest withdrawal's validator index
+        let latest_withdrawal = expected_withdrawals.last().expect("empty withdrawals");
+        let next_validator_index = (latest_withdrawal.validator_index + 1) % state.validators.len();
+        state.next_withdrawal_validator_index = next_validator_index;
+    } else {
+        // Advance sweep by the max length of the sweep if there was not a full set of withdrawals
+        let next_index =
+            state.next_withdrawal_validator_index + context.max_validators_per_withdrawals_sweep;
+        state.next_withdrawal_validator_index = next_index % state.validators.len();
+    }
+    Ok(())
+}
+
+pub fn get_expected_withdrawals<
+    const SLOTS_PER_HISTORICAL_ROOT: usize,
+    const HISTORICAL_ROOTS_LIMIT: usize,
+    const ETH1_DATA_VOTES_BOUND: usize,
+    const VALIDATOR_REGISTRY_LIMIT: usize,
+    const EPOCHS_PER_HISTORICAL_VECTOR: usize,
+    const EPOCHS_PER_SLASHINGS_VECTOR: usize,
+    const MAX_VALIDATORS_PER_COMMITTEE: usize,
+    const SYNC_COMMITTEE_SIZE: usize,
+    const BYTES_PER_LOGS_BLOOM: usize,
+    const MAX_EXTRA_DATA_BYTES: usize,
+>(
+    state: &BeaconState<
+        SLOTS_PER_HISTORICAL_ROOT,
+        HISTORICAL_ROOTS_LIMIT,
+        ETH1_DATA_VOTES_BOUND,
+        VALIDATOR_REGISTRY_LIMIT,
+        EPOCHS_PER_HISTORICAL_VECTOR,
+        EPOCHS_PER_SLASHINGS_VECTOR,
+        MAX_VALIDATORS_PER_COMMITTEE,
+        SYNC_COMMITTEE_SIZE,
+        BYTES_PER_LOGS_BLOOM,
+        MAX_EXTRA_DATA_BYTES,
+    >,
+    context: &Context,
+) -> Vec<Withdrawal> {
+    let epoch = get_current_epoch(state, context);
+    let mut withdrawal_index = state.next_withdrawal_index;
+    let mut validator_index = state.next_withdrawal_validator_index;
+    let mut withdrawals = vec![];
+    let bound = min(state.validators.len(), context.max_validators_per_withdrawals_sweep);
```

Comment:
should be able to drop the import up above for `std::cmp::min` and just do
```suggestion
    let bound = state.validators.len().min(context.max_validators_per_withdrawals_sweep);
```